subprojects/docs/src/docs/userguide/extending-gradle/task_configuration_avoidance.adoc

```diff
@@ -57,6 +57,25 @@ If you are configuring a task by name, you will need to use the configuration av
 
 In the event you still need to get access to a Task instance, you can use link:{javadocPath}/org/gradle/api/tasks/TaskCollection.html#named-java.lang.String-[TaskCollection.named(java.lang.String)] and call link:{javadocPath}/org/gradle/api/provider/Provider.html#get--[Provider.get()].  This will cause the task to be created/configured, but everything should work as it has with the eager APIs.
 
+[[sec:how_do_i_order_tasks]]
+=== How do I order tasks?
+
+In case, when task relationships needs to be established (`dependsOn`, `finalizedBy`, `mustRunAfter`, `shouldRunAfter`), configuration avoidance can also help to avoid configuring tasks.
+Calling these methods alone _will not cause task creation_, but it's important to understand their effects, as it might seem that they do so.
+
+[NOTE]
+====
+Calling these methods alone will _not_ cause task creation.
+They only declare relationships.
+====
+
+link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:mustRunAfter(java.lang.Object++[]++)[Task.mustRunAfter(...)] and link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:shouldRunAfter(java.lang.Object++[]++)[Task.shouldRunAfter(...)] represent soft relationships. These relationships will _order_ execution between the related tasks, but only if they exist in the same execution. They will not, in any context, cause task creation.
```

Comment:
Once you mention a "soft relationship", make it clearer that the next sentence in fact explains what that is. Ie. sneak the "meaning" word in there explicitly.

Reply:
```suggestion
link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:mustRunAfter(java.lang.Object++[]++)[Task.mustRunAfter(...)] and link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:shouldRunAfter(java.lang.Object++[]++)[Task.shouldRunAfter(...)] represent soft relationships. Soft relationships will _order_ execution between the related tasks; but only if they exist in the same execution. They will not, in any context, cause task creation.
```

Reply:
Tried to make a single sentence out of it, but it became eternal and hard to read.
Rather, I repeated the "soft" in the next sentence to repeat it.